dotCMS/src/main/java/com/dotcms/rest/api/v1/taillog/TailLogResource.java

```diff
@@ -0,0 +1,238 @@
+        final File logFile 	= new File(FileUtil.getAbsolutlePath(tailLogLofFolder + sanitizedFileName));
+
+
+        // if the logFile is outside of the logFolder, die
+        if ( !logFolder.exists() || !logFolder.canRead()
+                ||   !logFile.getCanonicalPath().startsWith(logFolder.getCanonicalPath())) {
+
+            final String errorMessage = "Invalid File request:" + logFile.getCanonicalPath() + " from:" +request.getRemoteHost();
+            SecurityLogger.logInfo(
+                    TailLogResource.class,  errorMessage);
+            return sendError(errorMessage);
+        }
+
+        Logger.info(this.getClass(), "Requested logFile:" + logFile.getCanonicalPath());
+
+        final String regex = Config.getStringProperty("TAIL_LOG_FILE_REGEX", ".*\\.log$|.*\\.out$");
+
+
+        if(!Pattern.compile(regex).matcher(sanitizedFileName).matches()){
+            return sendError("File name does not match a valid file pattern. ");
+        }
+
+        final MyTailerListener listener = new MyTailerListener();
+
+
+        final MyTailerThread thread = new MyTailerThread(logFile, listener, new EventOutput(),
+                linesBack);
+
+        String name = null;
+
+        //Finds a new thread to read the log
+        for (int i = 0; i < 1000; i++) {
+            name = "Log-Tailer" + i + ":" + sanitizedFileName;
+            Thread t = ThreadUtils.getThread(name);
+            if (t == null) {
+                Logger.warn(TailLogResource.class," TailLog Thread available with name: " +name);
+                break;
+            }
+            if (i > 100) {
+               return sendError("Too many Logger threads");
+            }
+        }
+
+        thread.setName(name);
+
+        thread.start();
+
+        return thread.getEventOutput();
+    }
+
+    private EventOutput sendError(final String errorMessage){
+
+        final EventOutput eventOutput = new EventOutput();
+
+        final OutboundEvent.Builder eventBuilder = new OutboundEvent.Builder();
+        eventBuilder.name("failure");
+        eventBuilder.data(Map.class,
+                map("failure", errorMessage));
+        final OutboundEvent event = eventBuilder.build();
+        try {
+            eventOutput.write(event);
+        } catch (Exception e1) {
+            throw new DotRuntimeException(e1);
+        }
+
+        return eventOutput;
+    }
+
+
+    static class MyTailerListener extends TailerListenerAdapter {
+
+        StringBuffer out = new StringBuffer();
+
+        public void handle(String line) {
+            getOut().append(UtilMethods.xmlEscape(line)).append("<br/>");
+        }
+
+        StringBuffer getOut() {
+            return out;
+        }
+
+        String getThenDispose() {
+            synchronized (this) {
+                try {
+                    return out.toString();
+                } finally {
+                    out.delete(0, out.length());
+                }
+            }
+        }
+
+    }
+
```

Comment:
So, StringBuilder is faster than StringBuffer.  Also, `setLength(0)` is more performant.  Additionally, I was wondering if we needed the sync on all the operations on the shared StringBuilder?
```
    static class MyTailerListener extends TailerListenerAdapter {

        private final StringBuilder out = new StringBuilder();

        public void handle(String line) {
            synchronized (this) {
                out.append(UtilMethods.xmlEscape(line)).append("<br/>");
            }
        }

        String getThenDispose() {
            synchronized (this) {
                try {
                    return out.toString();
                } finally {
                    out.setLength(0);
                }
            }
        }

    }
```

Reply:
dang I didn't know 'bout  that setLenght method

Reply:
I'll put it on a separate PR. thx